grblas/matrix.py

```diff
@@ -785,6 +785,39 @@ def apply(self, op, right=None, *, left=None):
             bt=self._is_transposed,
         )
 
+    def select(self, op, thunk=0):
+        """
+        GrB_Matrix_select
+        Compute IndexUnaryOp at each element of the calling Matrix, keeping
+        elements which return True.
+        """
+        method_name = "select"
+        op = get_typed_op(op, self.dtype, kind="indexunary")
+        self._expect_op(op, "IndexUnaryOp", within=method_name, argname="op")
+        if type(thunk) is not Scalar:
+            try:
+                thunk = Scalar.from_value(thunk, is_cscalar=None, name="")
+            except TypeError:
+                thunk = self._expect_type(
+                    thunk,
+                    Scalar,
+                    within=method_name,
+                    keyword_name="thunk",
+                    extra_message="Literal scalars also accepted.",
+                    op=op,
+                )
+        cfunc_name = f"GrB_Matrix_select_{self.dtype}"
```

Comment:
The dtype in the function is based on the dtype of the thunk value, right?  Also, this needs updated to handle UDTs.

Reply:
I'm not super-clear yet on the dtype handling, but the function with the dtype in the name is only for C scalars.  We should use `GrB_Matrix_select_Scalar` if the thunk is a GrB scalar.  I bet this would help the failing tests too.